grid-proxy/pkg/client/grid_client_test.go

```diff
@@ -270,3 +341,62 @@ func testSuccess(t *testing.T, f ProxyFunc) {
 		AssertHTTPRequest(t, f, endpoint.method, endpoint.path, endpoint.response, endpoint.call)
 	}
 }
+
+func TestPrepareURL(t *testing.T) {
+	status := "st"
+	freeMRU := uint64(10)
+	farmIDs := []uint64{1, 2, 3}
+	dedicated := true
+	filter := types.NodeFilter{
+		Status:    &status,
+		FreeMRU:   &freeMRU,
+		FarmIDs:   farmIDs,
+		Dedicated: &dedicated,
+	}
+	limit := types.DefaultLimit()
+
+	endpoint := "http://www.gridproxy.com"
+	client := Clientimpl{
+		endpoint: endpoint,
+	}
+
+	want := "http://www.gridproxy.com/nodes?status=st&free_mru=10&farm_ids=1&farm_ids=2&farm_ids=3&dedicated=true&size=50&page=1&ret_count=true&randomize=false"
+	wantURL, err := url.Parse(want)
+	assert.NoError(t, err)
+
+	got, err := client.prepareURL("nodes", filter, limit)
+	assert.NoError(t, err)
+
+	gotURL, err := url.Parse(got)
+	assert.NoError(t, err)
+
+	assert.Equal(t, wantURL.Path, gotURL.Path)
+	assert.True(t, reflect.DeepEqual(wantURL.Query(), gotURL.Query()))
+}
+
+func TestEncodeEmptyNodeFilter(t *testing.T) {
+	values := url.Values{}
```

Comment:
`var values url.Values`? please apply for others

Reply:
it needs to be initialized